src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -247,81 +249,51 @@ impl<T: Timestamp + Lattice + Codec64 + Display> AsyncStorageWorker<T> {
                                 .open(persist_location.clone())
                                 .await
                                 .expect("error creating persist client");
+                            let diagnostics = Diagnostics {
+                                shard_name: id.to_string(),
+                                handle_purpose: format!("resumption data {}", id),
+                            };
 
                             let mut write_handle = client
                                 .open_writer::<SourceData, (), T, Diff>(
                                     *data_shard,
                                     Arc::new(relation_desc.clone()),
                                     Arc::new(UnitSchema),
-                                    Diagnostics {
-                                        shard_name: id.to_string(),
-                                        handle_purpose: format!("resumption data {}", id),
-                                    },
+                                    diagnostics.clone(),
                                 )
                                 .await
                                 .unwrap();
                             let upper = write_handle.fetch_recent_upper().await;
+
+                            let read_handle = client
+                                .open_critical_since::<SourceData, (), T, Diff, PersistEpoch>(
+                                    *data_shard,
+                                    PersistClient::CONTROLLER_CRITICAL_SINCE,
+                                    diagnostics,
+                                )
+                                .await
+                                .expect("data shard");
+
+                            let export_as_of = {
```

Comment:
oh boy, I was so confused: sinks are also called exports in many parts. Totally not your fault, of course... 😅